core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -315,7 +314,7 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorAdded(executorId: String): Unit = synchronized {
     if (!executorIds.contains(executorId)) {
       executorIds.add(executorId)
-      executorIds.foreach(onExecutorIdle)
+      onExecutorIdle(executorId)
```

Comment:
The original rationale for marking other executors as idle is as follows. If an idle executor (call this executor X) is not removed because the minimum is reached, we need to mark it idle again when a new executor joins because we are no longer at the minimum. Otherwise, we may never removed executor X.

The issue here, however, is that executor X could be running a task, in which case it is no longer idle and we shouldn't mark it as such. This means we still need to mark it as idle **only** if it is not actually running anything. Does that make sense?